File: python-version/LC1760/minimum_size_test.py
import unittest
from minimum_size import Solution

class SolutionTest(unittest.TestCase):
    def test_minimumSize(self):
        solution = Solution()
        self.assertEqual(3, solution.minimumSize(nums = [9], maxOperations = 2))
        self.assertEqual(2, solution.minimumSize(nums = [2,4,8,2], maxOperations = 4))
        self.assertEqual(7, solution.minimumSize(nums = [7,17], maxOperations = 2))

if __name__ == '__main__':
    unittest.main()

File: go-version/leetcode/LC1760/minimum_size_test.go
package lc1760

import (
	"cutejiuge/leetcode-practice/util"
	"reflect"
	"testing"
)

type TestData map[string]struct {
	Nums          []int `yaml:"nums"`
	MaxOperations int   `yaml:"max_operations"`
	ExpectRes     int   `yaml:"expect_res"`
}

func TestMinimumSize(t *testing.T) {
	var testData TestData
	util.DataProvide(&testData)

	for caseName, caseData := range testData {
		res := minimumSize(caseData.Nums, caseData.MaxOperations)
		if !reflect.DeepEqual(res, caseData.ExpectRes) {
			t.Errorf("%s, res = %v, ExpectRes = %v", caseName, res, caseData.ExpectRes)
		}
	}
}
